apollo-server/v4/generated-types/src/index.ts

```diff
@@ -0,0 +1,45 @@
+import { ApolloServer } from '@apollo/server';
+import { startStandaloneServer } from '@apollo/server/standalone';
+// Here we import the automatically generated Book type, so we can use it in our
+// context typing.
+import { Book } from './__generated__/resolvers-types';
+import resolvers from './resolvers.js';
+import typeDefs from './schema.js';
+
+export interface MyContext {
+  dataSources: {
+    books: Book[];
+  };
+}
+
+// The ApolloServer constructor requires two parameters: your schema
+// definition and your set of resolvers.
+const server = new ApolloServer<MyContext>({
+  typeDefs,
+  resolvers,
+});
+
+// Our example static data set
+const books = [
+  {
+    title: 'The Awakening',
+    author: 'Kate Chopin',
+  },
+  {
+    title: 'City of Glass',
+    author: 'Paul Auster',
+  },
+];
+
+const { url } = await startStandaloneServer(server, {
+  context: async () => {
+    return {
+      // We are using a static data set for this example, but normally
+      // this would be where you'd add your data source connections
+      // or your REST API classes.
+      dataSources: { books },
```

Comment:
Even with the comment, this is still kinda confusing since it's very far from a convention we'd recommend (injecting static data into context). Could we make a very simple data source class that just returns the static book list and use that in our typical convention?

```ts
class BooksDataSource {
  books = [
    {
      title: 'The Awakening',
      author: 'Kate Chopin',
    },
    {
      title: 'City of Glass',
      author: 'Paul Auster',
    }
  ];

  async getBooks() {
    // simulate fetching a list of books
    return this.books;
  }

  async addBook(book: Book) {
    this.books.push(book);
    return {
      code: '200',
      success: true,
      message: 'New book added!',
      book: this.books[-1],
    };
  }
}
```

```ts
datasources: { books: new BooksDataSource() }
```

Reply:
Oh, this is such a great suggestion!! I did exactly this by adding another `datasources.ts` file.